services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx

```diff
@@ -46,6 +64,39 @@ export const SubmissionSummary = (): React.ReactElement => {
 
     if (error || !submission) return <GenericError />
 
+    const onUnlock = async () => {
+        try {
+            const result = await unlockStateSubmission({
+                variables: {
+                    input: {
+                        submissionID: submission.id
+                    }
+                }
+            })
+
+            if (result.errors) {
+                console.error('ERROR: got an error attempting to unlock', result.errors)
+                setUserVisibleUnlockError('Error attempting to unlock. Please try again.')
+                modalRef.current?.toggleModal(undefined, false)
+            }
+
+            if (result.data?.unlockStateSubmission.draftSubmission) {
+                console.log('Submission Unlocked')
+                modalRef.current?.toggleModal(undefined, false)
+            } else {
+                console.error('ERROR: Got nothing back from unlock')
+                modalRef.current?.toggleModal(undefined, false)
+                setUserVisibleUnlockError('Error attempting to unlock. Please try again.')
+            }
```

Comment:
I'm confused is this possible? I would hope unlock request would fail overall or return `.errors` if there is not a valid draft submission coming back. Can we push this error handling to backend

Reply:
OK, I wrote a wrapper for this controller. It would make sense to do the same on submit, but I’m going to leave that for another day.